### ISSUE TITLE
Catalog visibility: Hidden is not working, a bug perhaps?

### ISSUE THREAD

**Issue by AdriOosterwijk** [woocommerce/woocommerce#19172]
I discovered just now that the Catalog visibility: Hidden option is not working as it should (IMHO).
When set to hidden it still appears as a result when performing a search with the WordPress search widget. 
When clicking on this result I'm transferred to the single-product page and I am able to add it to my cart as if it is a "normal" product.

I checked if it was a theme related isuue but it also surfaces with a default WordPress theme (twentysixteen)
I checked if it was caused by another plugin by deactivating all of them but no avail so I think it may be a bug. I'm on a very tight schedule right now so I hope you are able to look into it very shortly.

I added a zip file with a few screenshots. I hope the file names are descriptive enough.

[Catalog_visibility_issue.zip](https://github.com/woocommerce/woocommerce/files/1767054/Catalog_visibility_issue.zip)

**Comment by mikejolley** [comment]
Hi there,

When you use WordPress search, the search results are formatted as blog posts. WooCommerce cannot control the output of this search.

If you use the WooCommerce search widget, search results are formatted like the rest of the product catalog, so these results are hidden as they should be.

Since the products are still published, they can be accessed directly and you shouldn't use 'hidden' as a secure method of hiding products from users. There will be links to them in places.

tl;dr if you want products to be truly hidden or private, use private status instead of publish status.

Thanks

**Comment by AdriOosterwijk** [comment]
Hey Mike,
That's weird. I just bought the Products Bundle plugin from WooCommerce.com. One of the features is to hide the single products from shop and search........ Form shop it works from search not. Makes the plugin next to useless. When using the WooCommerce search widget all other results are omitted....
Any pointers on how to solve it? I know this is not a support forum but still........

**Comment by mikejolley** [comment]
I'm sure when it says search it means product searching.

**Comment by AdriOosterwijk** [comment]
It seems to be but it is very confusing isn't it? At least IMHO. Now looking for a way to customize the search.

-> Omit results when it is post-type Product AND Catalog visibility is true. 

I'm not a coder you know. Great challenge.

**Comment by mikejolley** [comment]
I'd recommend not using regular WP search on a store, not if you want results to appear as products.

**Comment by AdriOosterwijk** [comment]
Recommendation noted it is just the way it should work in my case. Show all results (including products) when performing a search. Otherwise I will end up with a search field for Products and also a search field for anything else (blog, pages and such).

**Comment by AdriOosterwijk** [comment]
BTW the products may appear on the results page in the archive style.  As a matter of fact they do already.

**Comment by mikejolley** [comment]
And in those cases, WC cannot control the output of the results so hidden products cannot be hidden though the query. It's not the 'catalog'.

**Comment by truecounterfeit** [comment]
@mikejolley 
was this behavior resolved in version 5.0.3? 
* Fix - Component visibility cannot depend on matching conditions that reference currently hidden Components.

If not, is there a workaround?
